### PR TITLE
[New Feature]: Add support for transposing a tensor along a specified axes

### PR DESCRIPTION
**Pull request by silencrown**
## Description
Hello, while using TenSEAL, I noticed that the current version does not support transposing a tensor along a specified dimension. This may make it more complicated for users to implement functions like multi-head attention. Based on xtensor, I have modified the existing code to add support for the aforementioned operation in TenSEAL. In this code submission, users can utilize the Python method transpose(0, 2, 1). This operation is consistent with NumPy and can be adapted with minimal changes to support operations similar to PyTorch's transpose(-2, -1).

## Affected Dependencies
No

## How has this been tested?
The test code is included in this pull request.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by silencrown** [comment]
Here's a simple example that might better describe what this PR does. In the following example, the three operators `np.transpose(data, [0, 2, 1]` `torch_tensor.transpose(-2, -1)` `ckks_tensor.transpose([0, 2, 1]` are equivalent.

```python        
shape = [2, 3, 4]
expected_shape = [2, 4, 3]

data = np.arange(24).reshape(shape)

# numpy
np_trans = np.transpose(data, [0, 2, 1])
        
# torch
pt_trans = np.array(torch.tensor(data).transpose(-2, -1).tolist())

# CKKSTensor
enc_tensor = cxt_man.encrypt(data)
enc_trans = enc_tensor.transpose([0, 2, 1])
dec_trans = np.array(enc_trans.decrypt().tolist())

assert list(dec_trans.shape) == expected_shape
assert np.allclose(pt_trans, np_trans, rtol=0, atol=0.01)
assert np.allclose(dec_trans, np_trans, rtol=0, atol=0.01)
```
